src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supply = 0;
+        int buy = 0;
+        try (
+                BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                String operation = parts[0];
+                int amount = Integer.parseInt(parts[1]);
+
+                if ("supply".equals(operation)) {
+                    supply += amount;
+                } else if ("buy".equals(operation)) {
+                    buy += amount;
+                }
+            }
+            int result = supply - buy;
+
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+                writer.write("supply," + supply + "\n");
+                writer.write("buy," + buy + "\n");
+                writer.write("result," + result + "\n");
+            } catch (IOException e) {
+                System.out.println("Error occured while writing" + e.getMessage());
+            }
```

This exception handling violates checklist item #11. Simply printing to the console is not sufficient. You should re-throw the exception as a `RuntimeException`, including a descriptive message and the original exception as the cause. For example:
```java
} catch (IOException e) {
    throw new RuntimeException("Error occurred while writing to file " + toFileName, e);
}
```